maven-extension/src/main/java/io/opentelemetry/maven/OtelExecutionListener.java

```diff
@@ -200,11 +207,25 @@ public void mojoStarted(ExecutionEvent executionEvent) {
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_ID, mojoExecution.getExecutionId())
             .setAttribute(
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_LIFECYCLE_PHASE,
-                mojoExecution.getLifecyclePhase())
-            .startSpan();
+                mojoExecution.getLifecyclePhase());
+    //  enrich spans
+    List<MojoGoalExecutionHandler> handlers = getMojoGoalExecutionHandlers();
+    Optional<MojoGoalExecutionHandler> handler =
+        handlers.stream().filter(h -> h.supports(executionEvent)).findFirst();
+    logger.debug("OpenTelemetry: {} handler {}", executionEvent, handler);
+    handler.ifPresent(h -> h.enrichSpan(spanBuilder, executionEvent));
+
+    Span span = spanBuilder.startSpan();
     spanRegistry.putSpan(span, mojoExecution);
   }
 
+  /** TODO should we have a more generic and extensible mechanism to load handlers? */
```

Comment:
I think we should, as ideally we only attach the handlers that are applicable for a given build.

Reply:
I detailed more in the javadoc the options I identified. Questions I have are
* Will we ever have `MojoGoalExecutionHandlers` outside of this extension? Some could be added to the classpath but I wonder if it's realistic. I added such extension mechanisms on Jenkins plugins and it was never used outside of the library.
*  Will it have a performance impact? Probably not much.
* If we implement this, should we do it
   * "The Maven way" using the Maven Plexus container? I don't know how this would work and a quick Google search didn't help me
   * "The Otel way" using an SPI model with the `java.util.ServiceLoader`? Does it work well with the classloader isolation of Plexus? Probably yes.
*

Reply:
I'd lean towards the service loader approach, unless there are issues with Plexus classloaders.

For adding all the handlers, although there may not be much of a performance impact in calling handlers that aren't needed, I would be concerned about unintended outcomes if we took that approach.

Maybe I'm overthinking it and it isn't a problem, but if we got to a point of more than a couple of the handlers, I could see there being an impact on performance as it's called for each Mojo start event.

Reply:
I'll propose this